qulice-checkstyle/src/main/java/com/qulice/checkstyle/DiamondOperatorCheck.java

```diff
@@ -94,21 +94,22 @@ private static boolean isNotObjectBlock(final DetailAST node) {
     }
 
     /**
-     * Gets the return type of method or field type.
+     * Returns the first child node of a specified type, even
+     * in case when node contains as child an inner class type.
      *
      * @param node AST subtree to process.
      * @param type Type of token
      * @return Child node of specified type
      */
-    private static DetailAST getFirstChildNodeOfType(
+    private static DetailAST findFirstChildNodeOfType(
         final DetailAST node, final int type
     ) {
-        DetailAST result = node.getFirstChild();
-        if (result != null && result.getType() != type) {
-            final DetailAST child = DiamondOperatorCheck
-                .getFirstChildNodeOfType(result, type);
-            if (child == null) {
-                result = result.getNextSibling();
+        DetailAST result = node.findFirstToken(type);
+        if (result == null) {
+            final DetailAST dot = node.findFirstToken(TokenTypes.DOT);
```

Comment:
@mkordas  will think how to explain it better, basically I added this line with DOT, because findFirstToken works strange for me, it works not correctly when node tree contains inner declaration, probably it should not be explained in javadoc at all, what do you think